apps/citizen-func/utils/redis_storage.ts

```diff
@@ -4,67 +4,86 @@ import * as TE from "fp-ts/lib/TaskEither";
 import * as redis from "redis";
 
 export const setWithExpirationTask = (
-  redisClient: redis.RedisClientType,
+  redisClientTask: TE.TaskEither<Error, redis.RedisClientType>,
   key: string,
   value: string,
   expirationInSeconds: number,
   errorMsg?: string
-): TE.TaskEither<Error, true> =>
+): TE.TaskEither<Error, boolean> =>
   pipe(
-    TE.tryCatch(
-      () => redisClient.setEx(key, expirationInSeconds, value),
-      () =>
-        new Error(errorMsg ? errorMsg : "Error setting key value pair on redis")
+    redisClientTask,
+    TE.chain(client =>
+      TE.tryCatch(
+        () => client.setEx(key, expirationInSeconds, value),
+        () =>
+          new Error(
+            errorMsg ? errorMsg : "Error setting key value pair on redis"
+          )
+      )
     ),
-    TE.map(_ => true)
+    TE.map(constTrue)
```

Comment:
Can't we check if the response is `OK` or other?

Reply:
In case of error we would be in the left case i think

Reply:
You can see what I mean here https://github.com/pagopa/io-functions-cgn/blob/4df37c956d23a4aa2364a3017e4398bc71a81f18/utils/redis_storage.ts#L84 
Redis returns always something when in `right`.

Reply:
But be careful, it does not always return `OK`, sometimes it returns a number to indicate the number of touched records, and so on... just verify. The above example has some cases.

Reply:
Ok I get what you mean, according to the [doc](https://redis.io/docs/latest/commands/setex/) it seems to me that the return value should always be `OK` for the `setEx` method so I think that we can just use `constTrue` here, I can check for others ditto.
Do you agree?

Reply:
In theory, yes; but what if we get, for any reason, something different than `OK`? Are we checking it or just putting faith in it? `costTrue` is faith, an explicit check could be a good way to raise errors before finding yourself troubleshooting "why some records are not in redis and you didn't catch any error".
What do you think?

Reply:
Ok I agree, I'll add a check as you suggested

Reply:
[2270199](https://github.com/pagopa/io-messages/pull/12/commits/2270199ff45cfe82375a0f31e0540b3819f0aebb)